add: Новая болезнь после пересадки мозга

vovakons opened the pull request:
## Что этот ПР делает

Добавляет новую болезнь после пересадки мозга:
- проходит само через 10 минут
- сразу после дефиба дает минуту слепоты, глухоты, немоты и ползанья.
- 15% шанс на 10 окси урона и 2.5 урона мозгу
- 5% шанс на 3 секунды ползанья и 15 секунд запутанности ног
- 2.5% шанс на смерть случайного органа (*кроме мозга и сердца*), 15 секунд тряски, крик и 10 секунд ползанья, если же в крови находится новый реагент нейроматин, то будет только запутанность ног на 15 секунд с блевотой.

Новый реагент Нейроматин:
- передоз от 30 юнитов (*10% шанс на 15 секунд заутанности ног и блевота*)
- метаболизация 0.1 в тик
- лечит 0.5 урона мозгу в тик (*даже если мозг имеет урон больше 100*)
- уменьшает уровень бактерий на 1 в тик.
- стабилизирует сердечную недостаточность
- рецепт: sterilizine=3, mannitol=5, mitocholide=1 atropine=1 (нагреть до 350 К) выход 10 юнитов
- добавлено 3 автоиньъектора с новым реагентом во взломанный наномед плюс за 199 кредитов (инъектор на 15 юнитов)

## Почему это хорошо для игры

1. Добавляем дебафф самому легкому способу ревайва в игре, все остальные способы имеют свои минусы, а багаюзная пересадка не имеет - нехорошо.
2. Новый реагент для химиков
3. Врачам теперь надо будет следить за пациентом после пересадки, иначе он далеко не убежит с новой болезнью.

## Демонстрация изменений

<img width="492" height="475" alt="image" src="https://github.com/user-attachments/assets/908e787f-1981-4c1e-95b3-244d1ff68507" />


## Тестирование
Запустил на локалке, сделал пересадку - болезнь появляется.
Остальное тоже потестил, честно.

vovakons committed:
brain transplant decease

vovakons committed:
brain transpland disease fixes

vovakons committed:
neuromatin reagent

Новый реагент для ослабления симптома отказа органов у синдрома
пересадки мозга.
Добавлен новый инъектор в наномед плюс в категорию взломанных.

littleboobs commented on code/modules/reagents/chemistry/recipes/medicine.dm:
```diff
@@ -113,6 +113,15 @@
+	required_reagents = list("sterilizine" = 3, "mannitol" = 5, "mitocholide" = 1, "atropine" = 1)
+	result_amount = 10
+	min_temp = 350 //K
+	mix_message = "The mixture gives off a faint scent of almonds."
```
?

littleboobs commented on paradise.dme:
```diff
@@ -9,7 +9,6 @@
 // END_FILE_DIR
 // BEGIN_PREFERENCES
 #define DEBUG
-#define MULTIINSTANCE
```
????

Ksaikok46 commented on code/datums/diseases/brain_transplant_syndrome.dm:
```diff
@@ -0,0 +1,101 @@
+/datum/disease/brain_transplant_syndrome
+	name = "Синдром пересадки мозга"
+	agent = "Иммунитет"
+	desc = "Дизориентация, удушье, паралич и отказ органов."
```
```suggestion
	desc = "Дезориентация, удушье, паралич и отказ органов."
```

Ksaikok46 commented on code/datums/diseases/brain_transplant_syndrome.dm:
```diff
@@ -0,0 +1,101 @@
+	necrotize_organ_effect()
+
+/datum/disease/brain_transplant_syndrome/proc/necrotize_organ_effect()
+	to_chat(affected_mob, span_danger("Вы чувствуете, острую боль внутри!"))
```
```suggestion
	to_chat(affected_mob, span_danger("Вы чувствуете острую боль внутри!"))
```

Ksaikok46 commented on code/modules/reagents/chemistry/reagents/medicine.dm:
```diff
@@ -335,6 +335,34 @@
+/datum/reagent/medicine/neuromatin
+	name = "Нейроматин"
+	id = "neuromatin"
+	description = "лекарственное средство, рекомендуемое после трансплантации могза пациента."
```
```suggestion
	description = "лекарственное средство, рекомендуемое после трансплантации мозга пациента."
```

Ksaikok46 commented on code/datums/diseases/brain_transplant_syndrome.dm:
```diff
@@ -0,0 +1,101 @@
+	agent = "Иммунитет"
+	desc = "Дизориентация, удушье, паралич и отказ органов."
+	stage_prob = 10
+	cure_text = "Нейроматин (нейроадатик)"
```
Ниже там "нейроадаПтик", что наверное правильный вариант?
```suggestion
	cure_text = "Нейроматин (нейроадаптик)"
```

Ksaikok46 reviewed (commented):
Кажется бубс стареет

vovakons committed:
review fixes 1

vovakons committed:
review fixes 2

vovakons committed:
review fixes 3

ROdenFL commented on code/datums/diseases/brain_transplant_syndrome.dm:
```diff
@@ -0,0 +1,105 @@
+		heavy_effect()
+
+/datum/disease/brain_transplant_syndrome/proc/initial_effect()
+	affected_mob.Knockdown(60 SECONDS)
+	affected_mob.EyeBlind(60 SECONDS)
+	affected_mob.Deaf(60 SECONDS)
+	affected_mob.Stuttering(60 SECONDS)
```
сделай дефайн для продолжительности инит эффекта

ROdenFL commented on code/datums/diseases/brain_transplant_syndrome.dm:
```diff
@@ -0,0 +1,105 @@
+	if(!prob(BTS_LIGHT_EFFECT_PROBE))
+		return
+	affected_mob.emote("gasp")
+	to_chat(affected_mob, span_danger("Вы чувствуете, что вам не хватает воздуха!"))
```
когда угроза пользователю используется userdanger вообще

ROdenFL commented on code/datums/diseases/brain_transplant_syndrome.dm:
```diff
@@ -0,0 +1,105 @@
+	var/obj/item/organ/internal/select_organ = pick(organs)
+	select_organ.necrotize()
+
+/datum/disease/brain_transplant_syndrome/has_cure()
+	var/time_delta = world.time - start_time
+	return time_delta > BRAIN_TRANSPLANT_SYNDROME_DURATION
```
Без вызова родителя теряешь возможность лечится от нейроматина

vovakons replied:
так нейроматин только убирает отказ органов, лечения от болезни нет, проходит только само.

ROdenFL commented on code/datums/diseases/brain_transplant_syndrome.dm:
```diff
@@ -0,0 +1,105 @@
+  */
+/datum/disease/brain_transplant_syndrome
+	name = "Синдром пересадки мозга"
+	agent = "Иммунитет"
```
вот тут прикол не понял как иммунитет разносит болезнь

PlayerUnknown14 replied:
Именная система нового тела отторгает пересаженный мозг как инородный объект, пока он не приживётся. Наверное, я не медик.

PlayerUnknown14 commented on code/datums/diseases/brain_transplant_syndrome.dm:
```diff
@@ -0,0 +1,105 @@
+  */
+/datum/disease/brain_transplant_syndrome
+	name = "Синдром пересадки мозга"
+	agent = "Иммунитет"
```
```suggestion
	agent = "Иммунная система организма"
```

PlayerUnknown14 commented on code/datums/diseases/brain_transplant_syndrome.dm:
```diff
@@ -0,0 +1,105 @@
+  * - periodically oxy damage
+  */
+/datum/disease/brain_transplant_syndrome
+	name = "Синдром пересадки мозга"
```
```suggestion
	name = "Синдром трансплантации мозга"
```

PlayerUnknown14 commented on code/game/machinery/vending.dm:
```diff
@@ -2022,7 +2022,9 @@
-	contraband = list(/obj/item/reagent_containers/glass/bottle/sulfonal = 1, /obj/item/reagent_containers/glass/bottle/pancuronium = 1)
+	contraband = list(/obj/item/reagent_containers/glass/bottle/sulfonal = 1,
+					/obj/item/reagent_containers/glass/bottle/pancuronium = 1,
+					/obj/item/reagent_containers/hypospray/autoinjector/neuromatin = 3)
```
Цену на него сделай, да повыше. Достаточно сильный реагент, имхо.

PlayerUnknown14 commented on code/modules/reagents/chemistry/reagents/medicine.dm:
```diff
@@ -335,6 +335,34 @@
+/datum/reagent/medicine/neuromatin
+	name = "Нейроматин"
+	id = "neuromatin"
+	description = "лекарственное средство, рекомендуемое после трансплантации мозга пациента."
```
```suggestion
	description = "Лекарственное средство, предназначенное для противодействия последствиям трансплантации мозга. Имеет ряд побочных эффектов."
```

vovakons committed:
review fixes

ROdenFL commented:
Единственное, что смущает это нейроматин в списке лекарств у болезни, хотя он скорее симптомы лечит. В остальном ок вроде

PlayerUnknown14 committed:
Merge branch 'master220' into brain-transplant-debuff

PlayerUnknown14 commented on code/modules/reagents/chemistry/reagents/medicine.dm:
```diff
@@ -335,6 +335,34 @@
+/datum/reagent/medicine/neuromatin
+	name = "Нейроматин"
+	id = "neuromatin"
+	description = "Лекарственное средство, предназначенное для противодействия последствиям трансплантации мозга. Имеет ряд побочных эффектов."
```
```suggestion
	description = "Иммунодепрессант, предназначенный для противодействия последствиям трансплантации мозга. Имеет ряд побочных эффектов."
```

PlayerUnknown14 commented on code/modules/reagents/reagent_containers/hypospray.dm:
```diff
@@ -572,6 +572,23 @@
 
+/obj/item/reagent_containers/hypospray/autoinjector/neuromatin
+	name = "neuromatin autoinjector"
+	desc = "Маленький инъектор в форме ручки, содержащий внутри дозу нейроматина. Мощный нейроадаптик."
```
```suggestion
	desc = "Маленький инъектор в форме ручки, содержащий внутри дозу нейроматина. Мощный иммунодепрессант."
```

PlayerUnknown14 commented on code/datums/diseases/brain_transplant_syndrome.dm:
```diff
@@ -0,0 +1,108 @@
+  * - periodically oxy damage
+  */
+/datum/disease/brain_transplant_syndrome
+	name = "Синдром трансплантации мозга"
```
```suggestion
	name = "Синдром отторжения головного мозга"
```
Оригинальное название плохое, но конкретно в своём варианте особой уверенности нет. На твоё усмотрение.

PlayerUnknown14 commented on code/datums/diseases/brain_transplant_syndrome.dm:
```diff
@@ -0,0 +1,108 @@
+	agent = "Иммунная система организма"
+	desc = "Дезориентация, удушье, паралич и отказ органов."
+	stage_prob = 10
+	cure_text = "Нейроматин (нейроадаптик)"
```
```suggestion
	cure_text = "Нейроматин (иммунодепрессант)"
```

PlayerUnknown14 reviewed (changes requested):
Почитал, покумекал и словил мыслю назвать Нейроматин иммунодепрессантом. Ирл именно такие препараты применяют во время реабилитации после трансплантаций. Вообще нет уверенности, что такой вариант будет лучше, моё дело предложить. Решай сам, как по итогу сделать.

Мб в таком случае делать чела более восприимчивым к вирусам и инфекциям каким-то образом, но хз. Корованщиной уже отдаёт)))

vovakons committed:
neuromatin reagent desc update

Нейроматин переименован в иммунодепрессант.
Поменял название болезни на синдром отторжения головного мозга.

vovakons committed:
remove germs decrease from neuromatin

Убрал уменьшение бактерий в оргазиме после переименования в иммунодепрессант.

vovakons committed:
link brain to body for check transplantation

Добавлено поле в мозг чтобы хранить ссылку на оригинальное тело мозга.
При трансплантации и дефибе идет проверка тела мозга, если тело не
оригинальное - поле переписывается и выдается болезнь трансплантации.
Это сделано для того чтобы болезнь выдавалась при пересадке головы.

vovakons committed:
Merge branch 'master220' into brain-transplant-debuff

vovakons committed:
linter fix

dageavtobusnick committed:
Merge branch 'master220' into brain-transplant-debuff
